monkey/infection_monkey/plugin_compatability_verifier.py

```diff
@@ -0,0 +1,64 @@
+import logging
+from typing import Dict
+
+from common import HARD_CODED_EXPLOITER_MANIFESTS
+from common.agent_plugins import AgentPluginManifest, AgentPluginType
+
+from .i_puppet import IncompatibleOperatingSystemError
+from .island_api_client import IIslandAPIClient, IslandAPIError
+from .model import TargetHost
+
+logger = logging.getLogger(__name__)
+
+
+class PluginCompatabilityVerifier:
+    """
+    Verify plugin compatibility to run
+    """
+
+    def __init__(
+        self,
+        island_api_client: IIslandAPIClient,
+        exploiter_plugin_manifests: Dict[str, AgentPluginManifest] = HARD_CODED_EXPLOITER_MANIFESTS,
+    ):
+        self._island_api_client = island_api_client
+        self._exploiter_plugin_manifests = exploiter_plugin_manifests
+
+    def verify_exploiter_compatibility(self, exploiter_name: str, target_host: TargetHost):
+        """
+        Verify exploiter compatibility to run on a target host
+
+        It verifies based on a target host operating system and exploiter manifest
+        supported operating systems.
+        :param exploiter_name: Name of the exploiter
+        :param target_host: Target host
+        :raises IncompatibleOperatingSystemError: If exploiter is not compatible to run
+        """
+        exploiter_agent_plugin_manifest = self._get_exploiter_plugin_manifest(exploiter_name)
+        supported_exploiter_os = exploiter_agent_plugin_manifest.supported_operating_systems
```

Comment:
These names feel weird. Maybe just simply do:
```suggestion
        exploiter_plugin_manifest = self._get_exploiter_plugin_manifest(exploiter_name)
        supported_os = exploiter_agent_plugin_manifest.supported_operating_systems
```